endpoint.go

```diff
@@ -18,6 +18,8 @@ type Endpoint struct {
 	HTTPS bool
 	// set the flag true for skipping ssl verifications
 	Insecure bool
+        // if set, used to verify the hostname on the returned certificate
+        TLSServerName string
```

Comment:
There seem to have a whitespace issue there, can you run `make format`, then rebase an repush to this PR ?